### PR TITLE
Dealing with sous-amendements

### PR DESCRIPTION
**Pull request by davidbgk**

### CODE REVIEW

File: repondeur/zam_repondeur/models/amendement.py [hunk 1]
```diff
@@ -75,6 +75,7 @@
     Column("position", Integer, nullable=True),
     Column("discussion_commune", Integer, nullable=True),
     Column("identique", Boolean, nullable=True),
+    Column("parent", Text, nullable=True),
```
Comment (ronnix) [inline repondeur/zam_repondeur/models/amendement.py]: `num` est un `Integer` donc je me dis que ça aurait du sens d'utiliser ce même type pour `parent` ?

File: repondeur/zam_repondeur/writer.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def rename_field(field_name: str) -> str:
     return FIELDS_NAMES.get(field_name, field_name.capitalize())
 
 
-EXCLUDED_FIELDS = ["subdiv_contenu", "bookmarked_at"]
+EXCLUDED_FIELDS = ["subdiv_contenu", "bookmarked_at", "parent"]
```
Comment (ronnix) [inline repondeur/zam_repondeur/writer.py]: Peut-être que c'est une info valable à exporter ?

File: repondeur/zam_repondeur/models/amendement.py (hunk 1 above)
Comment (davidbgk) [reply repondeur/zam_repondeur/models/amendement.py]: C'est traitre car ça peut être `128 (Rect)` par exemple dans ce champ, le top ça serait de pouvoir faire un lien vers l'amendement à terme.

**Comment by ronnix** [comment]
Also, maybe it's time to add database migrations, so we don't have to wipe the DB every time we change the schema...

File: repondeur/zam_repondeur/models/amendement.py (hunk 1 above)
Comment (ronnix) [reply repondeur/zam_repondeur/models/amendement.py]: Dans ce cas là, on voudrait juste garder `128`. Peut-être qu'on peut utiliser `Amendement.parse_num()` ?

**Comment by ronnix** [comment]
Added migrations in #92 so when it's merged I think we can add a migration on this PR to add the extra column. (Sqlite has some limitations with schema changes, like renaming a column is unnecessarily complicated, but hopefully we won't run into that here...)

File: repondeur/zam_repondeur/fetch/models.py [hunk 3]
```diff
@@ -48,6 +50,20 @@ class Amendement:
 
     bookmarked_at: Optional[datetime] = None
 
+    @property
+    def parent(self) -> Optional["Amendement"]:
```
Comment (davidbgk) [inline repondeur/zam_repondeur/fetch/models.py]: @ronnix I'm pretty sure SQLAlchemy should do that for us, thoughts?

**Comment by davidbgk** [comment]
@ronnix r?

**Comment by ronnix** [comment]
Why do we need to store the `rectif` field ?

**Comment by davidbgk** [comment]
> Why do we need to store the `rectif` field ?

Mostly to display/query it correctly.

**Comment by ronnix** [comment]
Display I can understand, and maybe it’s a good reason to keep it.

I’m not sure about the query part though. See my question on Slack. If a parent amendement gets modified (its rectif increases), I think we should still consider the child amendement to be related to it. What do you think?

**Comment by davidbgk** [comment]
It's hard to choose without a real case and we'll adapt given available data, about the relation with self-reference, do you think we should keep it as is or use some kind of `relation`?

**Pull request merged**